lib/response.js

```diff
@@ -506,7 +506,22 @@ function patch(Response) {
                 // Finally, invoke the formatter and flush the request with it's
                 // results
 
-                return flush(self, formatter(self.req, self, body));
+                var formattedBody;
+                try {
+                    formattedBody = formatter(self.req, self, body);
+                } catch (e) {
+                    if (
+                        e instanceof errors.RestError ||
+                        e instanceof errors.HttpError
+                    ) {
+                        e.message =
```

Comment:
@hekike do you recall if we are using n/verror in restify core? Should we wrap this instead of modifying the existing message?

Reply:
NError via restify-errors https://github.com/restify/node-restify/blob/master/package.json#L111

Reply:
I think if you wrap you loose original statusCode and body of the error. Which is maybe fine here, just wanted to call out.

Reply:
Hmm, actually - thoughts on removing the try/catch in the formatter itself, but retaining the try/catch here? The InternalServerError wrapping could happen here and we can use NError to set the new error message too. I could see an argument for try/catch in the formatter _if_ the formatter had different types of errors, but I don't think we have a use case for that yet.

Reply:
The reasoning behind the try/catch in the formatter is opt-in/consent to an error being catchable. While exceptions in formatters should semantically always be catchable due to an expectation/reasonable assumption of statelessness, no such guarantee exists unfortunately. Coupling these w/n `node-restify` I think is perfectly reasonable, but since 3rd party formatters are supported, the formatter error/exception handling I think needs this 2-layer approach since otherwise true exceptions from 3rd party formatters will not properly be treated as exceptions.

Given the above, expecting error wrapping in the formatter seems the most pragmatic approach for differentiating catchable errors vs true exceptions, and given this expectation of wrapping, additional wrapping w/n `response.js` seems superfluous.

Reply:
Resolving with @hekike's recommendation to not mutate the error:
https://github.com/restify/node-restify/pull/1807/commits/84e456a066543731dd6e01bccc92f0b5af6cbe25